src/openrct2/network/Http.h

```diff
@@ -0,0 +1,86 @@
+#pragma region Copyright(c) 2014 - 2018 OpenRCT2 Developers
+/*****************************************************************************
+ * OpenRCT2, an open source clone of Roller Coaster Tycoon 2.
+ *
+ * OpenRCT2 is the work of many authors, a full list can be found in contributors.md
+ * For more information, visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * A full copy of the GNU General Public License can be found in licence.txt
+ *****************************************************************************/
+#pragma endregion
+
+#ifndef OPENRCT2_NETWORK_HTTP_H_
+#define OPENRCT2_NETWORK_HTTP_H_
+
+#ifndef DISABLE_HTTP
+
+#include <atomic>
+#include <functional>
+#include <map>
+#include <string>
+
+namespace http
+{
+
+enum class Status
+{
+    OK = 200
+};
+
+enum class Method
+{
+    GET,
+    POST,
+    PUT
+};
+
+struct Response
+{
+    Status                             status;
+    std::string                        content_type;
+    std::string                        body;
+    std::map<std::string, std::string> header;
+    std::string                        error;
+};
+
+struct Request
+{
+    std::string                        url;
+    std::map<std::string, std::string> header;
+    Method                             method;
+    std::string                        body;
+};
+
+struct Client
+{
+    Client();
+    ~Client();
+
+    Response Do(Request req);
+
+private:
+    static std::atomic<int> refcount;
```

Comment:
Avoid `int` as it is not guaranteed to be the same size on all platforms.

Reply:
Use our typedefed types instead:

https://github.com/OpenRCT2/OpenRCT2/blob/e602162fdb6ad2041d236ca10344ef3366bd895b/src/openrct2/common.h#L38-L45